### PR TITLE
Implementing enable_singlepass_benchmarks lint

### LINKED ISSUE #11: Add the enable-singlepass-benchmarks lint to verify that benchmarks are compiled and run during unit tests (single pass).
In Substrate, benchmarks are not automatically run when running cargo test, meaning that PRs may be mergable if only `cargo test` is used to verify the codebase. It is quite trivial to configure benchmarks to run during testing (we do it in every Composable pallet), so a lint that ensures that this method is used would be useful.

### PR DESCRIPTION
**Pull request by TheCodingWombat**
Implementing enable_singlepass_benchmarks lint using `ripgrep`. Catches patterns like
`#[cfg(feature="runtime-benchmarks")]`
`#[cfg(any(feature="runtime-benchmarks", other_stuff="other_stuff"))]`
and suggests to also compile during `test`.

Includes tests.

Does not include auto-fix.

Closes #11

### CODE REVIEW

**Comment by TheCodingWombat** [comment]
Problem: Current suggestions overwrite other cfg attributes with test attribute.

**Comment by TheCodingWombat** [comment]
Problem fixed.

File: substrace_lints/src/substrace_lints/enable_singlepass_benchmarks.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+fn create_suggested_text(text: &str) -> String {
+    if text.contains("any(") { // Example: #[cfg(any(feature = "runtime-benchmarks"))]
+        text.replacen("any(", "any(test, ", 1)
+    } else { // Example: #[cfg(feature = "runtime-benchmarks")]
+        String::from("#[cfg(any(test, feature = \"runtime-benchmarks\"))]")
+    }
+}
+
+// Run ripgrep with the supplied pattern and path and output to jsonl format.
+fn run_ripgrep(pattern: &str, path: &str) -> std::process::Output {
```
Comment (KaiserKarel) [inline substrace_lints/src/substrace_lints/enable_singlepass_benchmarks.rs]: This requires ripgrep to be installed. Can we instead link ripgrep in substrace using https://docs.rs/grep-searcher/0.1.8/grep_searcher/index.html?

**Review by KaiserKarel — CHANGES_REQUESTED** [review]
I'd prefer using ripgrep as a library

File: substrace_lints/src/substrace_lints/enable_singlepass_benchmarks.rs (hunk 1 above)
Comment (TheCodingWombat) [reply substrace_lints/src/substrace_lints/enable_singlepass_benchmarks.rs]: Now using grep library with pcre2 regex matcher crate

**Review by TheCodingWombat — COMMENTED** [review]
Now using grep library

File: substrace_lints/src/substrace_lints/enable_singlepass_benchmarks.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    data: serde_json::Value,
+}
+
+//TODO: Implement auto-fix.
+impl<'tcx> LateLintPass<'tcx> for EnableSinglepassBenchmarks {
+
+    fn check_crate(&mut self, cx: &LateContext<'tcx>) {
+
+        //TODO: Disregard patterns matched in block comments
+        // See this pattern in action: https://regex101.com/r/jd6CX1/4
```
Comment (KaiserKarel) [inline substrace_lints/src/substrace_lints/enable_singlepass_benchmarks.rs]: I don't mind this catching errors in comments; that enforces best practices in comments too.

**Review by KaiserKarel — APPROVED** [review]

**Pull request merged**